shared/Microsoft.Extensions.Buffers.MemoryPool.Sources/SlabMemoryPool.cs

```diff
@@ -119,6 +127,7 @@ private MemoryPoolBlock AllocateSlab()
                 block.IsLeased = true;
 #endif
                 Return(block);
+                blocksAllocated++;
```

Comment:
We create additional segment after the loop. It takes the last `_blockSize` of memory.

Reply:
Ahhhh. I think it's technically right then but still a little hard to follow. I think it would be a lot easier to grok (if only because less bitwise operations) if written as follows:

```c#
// ...

var maxPenultimateOffset = _slabLength - (_blockSize * 2);
var offset = firstOffset;
for (;
    offset <= maxPenultimateOffset;
    offset += _blockSize)
{
    var block = new MemoryPoolBlock(
        this,
        slab,
        offset,
        _blockSize);
    
    // ...
```